both/methods.js

```diff
@@ -153,16 +153,25 @@ Meteor.methods({
 
     Meteor.call('refreshRoundCodes', competitionId, eventCode);
   },
-  addNonEventRound: function(competitionId, round) {
+  addScheduleEvent: function(competitionId, eventData, roundId) {
     check(competitionId, String);
     throwIfCannotManageCompetition(this.userId, competitionId);
-    return Rounds.insert({
+
+    var round = Rounds.findOne(roundId);
+
+    return ScheduleEvents.insert({
       competitionId: competitionId,
-      title: round.title,
-      startMinutes: round.startMinutes,
-      durationMinutes: round.durationMinutes,
+      roundId: (round ? round._id : null),
+      title: (round ? round.displayTitle() : eventData.title),
```

Comment:
Hm... Sounds very "reactive", doesn't it?

Meteor is so good at propagating DB changes to browser windows everywhere. I wish there was a similar mechanism to propagate them to other parts of the DB.

You could also imagine computed fields in the DB. Don't know if Mongo has anything like that.

This is not hard to do in any number of ways, but I'm curious what the best pattern is.